skimage/future/graph/rag.py

```diff
@@ -461,20 +457,32 @@ def draw_rag(labels, rag, img, border_color=None, node_color='#ffff00',
     >>> img = data.coffee()
     >>> labels = segmentation.slic(img)
     >>> g =  graph.rag_mean_color(img, labels)
-    >>> out = graph.draw_rag(labels, g, img)
+    >>> smap = graph.show_rag(labels, g, img)
+    >>> cbar = plt.colorbar(smap)
     """
+
     if not in_place:
         rag = rag.copy()
 
-    if desaturate:
-        img = color.rgb2gray(img)
-        img = color.gray2rgb(img)
-
+    if ax is None:
+        fix, ax = plt.subplots()
     out = util.img_as_float(img, force_copy=True)
-    cc = colors.ColorConverter()
 
-    edge_color = cc.to_rgb(edge_color)
-    node_color = cc.to_rgb(node_color)
+    if img_cmap is None:
+        if img.ndim < 3:
+            raise ValueError('The image needs to be RGB')
+        if img.ndim < 4:
+            r, c, _ = img.shape
+            new_img = np.empty((r, c, 4), dtype=np.float)
+            new_img[:, :, 0:3] = img
+            new_img[:, :, 3] = 1
+            out = new_img
+    else:
+        img_cmap = cm.get_cmap(img_cmap)
+        out = color.rgb2gray(img)
+        out = img_cmap(out)
+
+    edge_cmap = cm.get_cmap(edge_cmap)
```

Comment:
I think we've discussed this, but can you remind me again, why does this work when `edge_cmap` is `'viridis'` and the matplotlib version is < 1.5?